src/ui/interaction/files/AvFileUpload/AvFileUpload.vue

```diff
@@ -0,0 +1,267 @@
+   * If not specified, a random ID is generated.
+   *
+   * @default () => useRandomId(...)
+   */
+  id?: string
+
+  /**
+   * ARIA label for file download button.
+   *
+   * @default ''
+   */
+  ariaLabel?: string
+
+  /**
+   * Accepted file types, specified as a string (like HTML `accept` attribute)
+   * or an array of strings (which will be transformed into a string).
+   *
+   * @default undefined
+   */
+  accept?: string | string[]
+
+  /**
+   * Error message to be displayed in case of download problem.
+   *
+   * @default ''
+   */
+  error?: string
+
+  /**
+   * Message indicating that the downloaded file is valid.
+   *
+   * @default ''
+   */
+  validMessage?: string
+
+  /**
+   * Whether the file upload input is disabled.
+   *
+   * @default false
+   */
+  disabled?: boolean
+
+  /**
+   * Value linked to file upload input template.
+   *
+   * @default ''
+   */
+  modelValue?: string
+}
+
+defineOptions({
+  inheritAttrs: false,
+})
+
+const props = withDefaults(defineProps<AvFileUploadProps>(), {
+  id: () => useRandomId('file-upload'),
+  ariaLabel: '',
+  accept: undefined,
+  validMessage: '',
+  error: '',
+  modelValue: '',
+})
+
+/**
+ * Emits available for the AvFileUpload component.
+ *
+ * @emits update:modelValue - Event emitted when the model value linked to the file is updated.
+ * @param payload - The updated model value (string).
+ *
+ * @emits change - Event emitted when the selected file is changed.
+ * @param payload - The new list of selected files (FileList).
+ */
+const emit = defineEmits<{
+  /**
+   * Event emitted when the model value linked to the file is updated.
+   * @param payload The updated model value (string).
+   */
+  (e: 'update:modelValue', payload: string): void
+
+  /**
+   * Event emitted when the selected file is changed.
+   * @param payload The new list of selected files (FileList).
+   */
+  (e: 'change', payload: FileList): void
+}>()
+
+defineSlots<{
+  /**
+   * Slot for the hint description.
+   */
+  hint?: Slot
+
+  /**
+   * Default slot for global content between the left and right icons.
+   */
+  default?: Slot
+}>()
+
+const acceptTypes = computed(() => {
+  if (Array.isArray(props.accept)) {
+    return props.accept.join(',')
+  }
+  return props.accept
+})
+
+const isDragging = ref(false)
+
+function onDrop (event: DragEvent) {
+  event.preventDefault()
+  isDragging.value = false
+
+  if (!props.disabled && event.dataTransfer?.files?.length) {
+    emit('change', event.dataTransfer.files)
+    emit('update:modelValue', event.dataTransfer.files[0]?.name ?? '')
+  }
+}
+
+function onDragOver (event: DragEvent) {
+  event.preventDefault()
+  if (!props.disabled) {
+    isDragging.value = true
+  }
+}
+
+function onDragLeave () {
+  isDragging.value = false
+}
+
+function onChange ($event: InputEvent) {
+  emit('update:modelValue', ($event.target as HTMLInputElement)?.value)
+  emit('change', ($event.target as (InputEvent['target'] & { files: FileList }))?.files)
```

Comment:
ça doit être qlq chose comme ça

```suggestion
const input = event.target as HTMLInputElement
  const files = Array.from(input.files || [])
  emit('update:modelValue', files)
  emit('change', files)
```

Reply:
https://vue-ds.fr/composants/DsfrFileUpload#%E2%9A%99%EF%B8%8F-code-source-du-composant

<img width="700" height="126" alt="image" src="https://github.com/user-attachments/assets/402388eb-8a35-41de-87b9-ac3022097f8e" />

Reply:
C'est juste que je ne vois pas l'interet d'emit le nom du fichier